platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/AddressBookInitializer.java

```diff
@@ -97,31 +102,30 @@ public class AddressBookInitializer {
      * @param signedState          The signed state loaded from disk.  May be null.
      * @param genesisSupplier      The swirld application state in genesis start. Must not be null.
      * @param configAddressBook    The address book derived from config.txt. Must not be null.
-     * @param parentDirectory      The parent directory of the address book directory. Must not be null.
-     * @param useConfigAddressBook Indicates if the unmodified config address book should be used.
+     * @param addressBookConfig    The configuration settings for AddressBooks.
      */
     public AddressBookInitializer(
             @NonNull final SoftwareVersion currentVersion,
             @Nullable final SignedState signedState,
             @NonNull final Supplier<SwirldState> genesisSupplier,
             @NonNull final AddressBook configAddressBook,
-            @NonNull final String parentDirectory,
-            final boolean useConfigAddressBook) {
+            @NonNull final AddressBookConfig addressBookConfig) {
         this.currentVersion = Objects.requireNonNull(currentVersion, "The currentVersion must not be null.");
         this.genesisSupplier =
                 Objects.requireNonNull(genesisSupplier, "The genesis swirldState supplier must not be null.");
         this.configAddressBook = Objects.requireNonNull(configAddressBook, "The configAddressBook must not be null.");
-        Objects.requireNonNull(parentDirectory, "The parentDirectory must not be null.");
-        this.pathToAddressBookDirectory = Path.of(parentDirectory, ADDRESS_BOOK_DIRECTORY_NAME);
+        Objects.requireNonNull(addressBookConfig, "The addressBookConfig must not be null.");
+        this.loadedSignedState = signedState;
+        this.loadedAddressBook = loadedSignedState == null ? null : loadedSignedState.getAddressBook();
+        this.pathToAddressBookDirectory = Path.of(addressBookConfig.addressBookDirectory());
```

Comment:
With this change, `ADDRESS_BOOK_DIRECTORY_NAME` is no longer used

Reply:
removed in next commit.